lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
+        end
 
-    def sanitize_filename(filename)
-      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
-      sanitized_filename = sanitized_filename.gsub('index.html', '')
-      sanitized_filename.gsub('.html', '')
+        Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+        File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
+      end
     end
 
-    def generateSubgroup(filename, key, value, all_subgroups, level)
-      title = value[:title].split('–').last
-      
-      subsubgroup_list = all_subgroups.select do |subsubgroup_key, _subsubgroup_value|
-        subsubgroup_key.include? key and subsubgroup_key != key and \
-          key.split('/').length > level
+    def generate_sub_group(filename, path, page, all_child_pages, level)
+      title = page[:title].split('–').last.strip
+
+      url = page[:url]
+      headers = page[:headers]
+
+      grandchildren = all_child_pages.select do |grandchild_path, _|
+        grandchild_path.include? path and grandchild_path != path and \
+          path.split('/').length > level
       end
 
-      subgroup = ''
-      has_subgroups = !all_subgroups.empty?
-      if value[:headers].any? || !subsubgroup_list.empty?
-        if has_subgroups
-          url = value[:url]
-          active = active?(filename, url, true)
-          # puts "#{url}, #{filename}, #{key}" if active
-          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
-          subgroup << "<li class=\"#{item_class}\">"
-          subgroup << "<div class=\"nav-subgroup-heading\"><i class=\"material-icons\">arrow_right</i><a href=\"#{url}\">#{title}</a></div>"
-          subgroup << '<ul class="nav-ul">'
+      sub_group_markup = ''
+      has_child_pages = !all_child_pages.empty?
 
-          if subsubgroup_list.empty?
-            value[:headers].each do |header|
-              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+      if (!headers.nil? && headers.any?) || !grandchildren.empty?
+        if has_child_pages
+          active = filename.active?(url, exact: true)
+          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
+          sub_group_markup << "<li class=\"#{item_class}\">"
+          sub_group_markup << '<div class="nav-subgroup-heading">'
+          sub_group_markup << '<i class="material-icons">arrow_right</i>'
+          sub_group_markup << "<a href=\"#{url}\">#{title}</a>"
+          sub_group_markup << '</div>'
+          sub_group_markup << '<ul class="nav-ul">'
+
+          if grandchildren.empty?
+            headers.each do |header|
+              hash = header[:hash]
+              subtitle = header[:title]
+              sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
```

Comment:
Codacy found an issue: [Line is too long. [105/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)